### PR TITLE
DRAFT: ensure by default a non-admin user can NOT restore a backup

### PR DESCRIPTION
**Pull request by weshayutin**
* Due to the potential of a non-admin user restoring a backup to an inappropriate namespace the restore resources have been removed by default.

* Remove create access for BSL to non-admin users

### CODE REVIEW

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml [hunk 1]
```diff
@@ -17,9 +23,21 @@ objects:
...
   - apiGroups:
+    # By default restore resources will NOT be provided by this template.
+    # It is recommended that restores are executed by OpenShift Admin's in production environments.
+    # For demonstration purposes only you may uncomment the restore resources.
     - velero.io
     resources:
-    - '*'
+    - 'backuprepositories'
+    - 'backups'
+    - 'deletebackuprequests'
```
Comment (shawn-hurley) [inline blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:33]: Probably just want admins , some one may be able to delete someone else's?

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml [hunk 2]
```diff
@@ -17,9 +23,21 @@ objects:
     name: oadp-nonadmin-role
   rules:
   - apiGroups:
+    # By default restore resources will NOT be provided by this template.
+    # It is recommended that restores are executed by OpenShift Admin's in production environments.
+    # For demonstration purposes only you may uncomment the restore resources.
     - velero.io
     resources:
-    - '*'
+    - 'backuprepositories'
```
Comment (shawn-hurley) [inline blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:31]: Do we want these users to be able to create a new back repo they control them backup the whole cluster?

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml [hunk 3]
```diff
@@ -17,9 +23,21 @@ objects:
...
+    - 'backups'
+    - 'deletebackuprequests'
+    # - 'downloadrequests'
+    - 'podvolumebackups'
+    - 'podvolumerestores'
```
Comment (shawn-hurley) [inline blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:36]: Feel like this should be same as restores

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml [hunk 4]
```diff
@@ -17,9 +23,21 @@ objects:
...
+    # - 'restores'
+    # - 'schedules'
+    - 'serverstatusrequests'
+    - 'volumesnapshotlocations'
```
Comment (shawn-hurley) [inline blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:40]: Same reason as backup storage locations

**Review by shawn-hurley — COMMENTED** [review]
Just wondering about other resources

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml (hunk 2 above)
Comment (weshayutin) [reply blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:31]: backuprepositories are created per restic backup correct? This is not user defined iiuc and would be needed for any restic backup. Maybe I'm wrong

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml (hunk 4 above)
Comment (weshayutin) [reply blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:40]: bah.. too fast. thanks

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml (hunk 3 above)
Comment (weshayutin) [reply blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:36]: aye of course.. too fast

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml (hunk 1 above)
Comment (weshayutin) [reply blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:33]: meh.. it's just a guide.. so I'll remark it out

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml (hunk 2 above)
Comment (sseago) [reply blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:31]: @shawn-hurley Backup repositories are created by velero for restic/kopia/datamover backups. End users never create them (although for debugging purposes, sometimes they must remove them, but I'm assuming that can remain for admins only).

Comment (sseago) [reply blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:31]: But I don't think the non-admin user or tekton needs any permissions around this -- the velero pod will create it as needed.

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml (hunk 3 above)
Comment (sseago) [reply blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml]: Hmm. But again, there is a difference. The tekton/non-admin application will need to create backups and restores, but won't need to create PVBs/PVRs, so the permission level for this application could be lower for those.

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml (hunk 2 above)
Comment (kaovilai) [reply blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:31]: >backuprepositories are created per restic backup correct? 

Per backupStorageLocation.

Comment (sseago) [reply blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:31]: @kaovilai also per namespace.

File: blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml [hunk 5]
```diff
@@ -17,9 +23,20 @@ objects:
...
+    # - 'deletebackuprequests'
+    # - 'downloadrequests'
+    - 'podvolumebackups'
+    # - 'podvolumerestores'
+    # - 'restores'
+    # - 'schedules'
+    - 'serverstatusrequests'
     verbs:
     - create
     - delete
```
[the remark this answers is not in the document]
Comment (kaovilai) [reply blogs/tekton-oadp-nonadmin/install_templates/templates/03-rbac-pipeline-role.yaml:42]: Would update leak info?

**Pull request closed without merging**